tests/prague/eip2537_bls_12_381_precompiles/helpers.py

```diff
@@ -61,3 +65,579 @@ def vectors_from_file(filename: str) -> List:
+            1731081574565817469321317449275278355306982786154072576198758675751495027640363897075486577327802192163339186341827,
+        ]
+        return [FP(t) for t in kernel_ts]
+
+    @classmethod
+    def generate_g2_kernel_points(cls) -> FP2:
+        """
+        Return precomputed kernel points for the BLS12-381 G1 curve, that map to the g2 identity
+        point. These are generated using sage math externally with the following script as its
+        significantly faster than using `py_ecc` (200-1000x faster).
+
+        Note we cannot use sage math directly in the code as it is not a pure Python library and
+        requires an external dependency to be installed on the system machine.
+
+        ```sage
+        q = 0x1A0111EA397FE69A4B1BA7B6434BACD764774B84F38512BF6730D2A0F6B0F6241EABFFFEB153FFFFB9FEFFFFFFFFAAAB
+        Fp = GF(q)
+        R.<x> = PolynomialRing(Fp)
+        Fp2.<u> = Fp.extension(x^2 + 1)
+        E2 = EllipticCurve(Fp2, [0, 4*(1+u)]) # BLS12-381 G2 curve
+        ISO_3_A = 240 * u
+        ISO_3_B = 1012 * (1 + u)
+        ISO_3_Z = -(2 + u)
+        Ei = EllipticCurve(Fp2, [ISO_3_A, ISO_3_B])
+        iso = EllipticCurveIsogeny(E=E2, kernel=None, codomain=Ei, degree=3).dual()
+        for (x, _) in iso.kernel_polynomial().roots():
+            inv_factor = (x * ISO_3_A / -ISO_3_B) - 1
+            if inv_factor == 0:
+                continue
+            discriminant = 1 + 4 / inv_factor
+            if not discriminant.is_square():
+                continue
+            for sign in [1, -1]:
+                zt2 = (-1 + sign * discriminant.sqrt()) / 2
+                t2 = zt2 / ISO_3_Z
+                if t2.is_square():
+                    t = t2.sqrt()
+                    x1 = -ISO_3_B / ISO_3_A * (1 + 1 / (ISO_3_Z^2 * t^4 + ISO_3_Z * t^2))
+                    assert x1 == x, f"Verification failed: x1={x1} != x={x}"
+                    print(t)
+        ```
+
+        Add the script contents to a file called `points.sage`, run `sage points.sage`!
+
+        Please see the sage math installation guide to replicate:
+            - https://doc.sagemath.org/html/en/installation/index.html
+        """  # noqa: E501
+        # There are only 3 points for the degree 3 isogeny kernel:
+        # 1. The point at infinity (doesn't apply here as we can't represent it as an Fp2
+        #    field element - it's a curve point concept.
+        # 2. A non-trivial point P (what we generated with the sage script).
+        # 3. The point -P (negation of P).
+        #
+        # Note no field elements in FP2 map to P or -P through the SWU map!!
+        return FP2(
+            (
+                1321391878976200074119645034444121025665939532471580297403784397444487204385014689930429494040875145883332046150361,
+                2445707658947438392540268931313688646028792845302972805205282500613101640661654934098821383626938971038899010733909,
+            ),
+        )
```

Comment:
I'm thinking that for the time being this should return an empty list and we should make a comment that the value returned by the algorithm for some reason do not result in the `INF_G2` response we expect.